src/mainwindow.h

```diff
@@ -141,6 +143,8 @@ private slots:
 
     HashTypeChecker m_hashTypeChecker;
     JohnHandler     m_johnGuess;
+    JohnAttack      m_johnDefaultFormat;
+    QString         m_defaultFormat;
```

Comment:
Can't we reformat the code to be less dummy and get rid of both m_format and m_defaultFormat ?
Hint: How about setting the default format for a session in the JohnSession?